examples/examples/z_pub_thr.rs

```diff
@@ -40,10 +53,28 @@ fn main() {
     let mut count: usize = 0;
     let mut start = std::time::Instant::now();
     loop {
-        publisher.put(data.clone()).res().unwrap();
+        let attachments = (args.attachments_number != 0).then(|| {
```

Comment:
I suggest to remove the `attachments` from the `z_pub_thr` and to have it only on the `z_pub/z_sub`.

Reply:
`z_sub` example should also show how to use the attachment.

Reply:
@Mallets in [zenoh-c](https://github.com/eclipse-zenoh/zenoh-c/pull/203/files) I decided to move attachments to separate sample to don't overload main simple z_pub/z_sub samples.
Maybe we should do the same here too?

Reply:
I'm fine to move it to separate examples.